backend/molgenis-emx2-rdf/src/main/java/org/molgenis/emx2/rdf/RDFService.java

```diff
@@ -538,7 +536,35 @@ private List<IRI> getIriValue(final Row row, final Column column) {
 
     for (final var item : items.values()) {
       PrimaryKey key = new PrimaryKey(item);
-      iris.add(Values.iri(ns, tableName + "/" + key.getEncodedValue()));
+      final TableMetadata target = column.getRefTable();
+      final Database database = target.getSchema().getDatabase();
+
+      // If the table has a mg_tableclass column then the row might actually be from a child table
+      if (target.getColumnNames().contains("mg_tableclass")) {
+        // Retrieve the rows mg_tableclass to figure out the child table.
+        var query = target.getTable().query();
+        query.select(s("mg_tableclass"));
+        query.where(key.getFilter());
+        for (var referencedRow : query.retrieveRows()) {
```

Comment:
this looks expensive to me; if I understand correctly this results in a query per ref item?

I think we should consider that the original query already includes this information somehow. I mean we could request a 'nameoftherefcolumn.mg_tableclass'

Reply:
Thinking of it a bit more I think we should do this in the getRows method and not in the underlying sql package, unless we have more use cases where we need it. Because if we would do it in the underlying sql package then you would always carry the cost of joining tables.